segnet/batch_inference.py

```diff
@@ -0,0 +1,119 @@
+# https://github.com/BVLC/caffe/issues/861
+import matplotlib
+matplotlib.use('Agg')
+
+import re
+import os
+import os.path
+import sys
+import requests
+import numpy as np
+from PIL import Image
+import click
+import json
+import StringIO
+import tempfile
+from boto3.session import Session
+caffe_root = os.getenv('CAFFE_ROOT', '/home/ubuntu/caffe-segnet/')
+sys.path.insert(0, caffe_root + 'python')
+import caffe
+
+from inference import predict
+from queue import receive
+
+aws_session = Session()
+s3 = aws_session.client('s3')
+
+
+def parse_s3_uri(s3uri):
+    match = re.search('s3://([^/]*)/(.*)$', s3uri)
+    if not match:
+        return None
+    return (match.group(1), match.group(2))
+
+
+def resolve_s3(s3uri):
+    parsed = parse_s3_uri(s3uri)
```

Comment:
Yeah, definitely -- similar with the setup_net function, and maybe use `click` over there also.